src/bgw/job_stat.c

```diff
@@ -636,18 +638,26 @@ ts_bgw_job_stat_mark_start(int32 bgw_job_id)
 	 * requires a lock that it not NoLock). */
 	Relation rel =
 		table_open(catalog_get_table_id(ts_catalog_get(), BGW_JOB_STAT), ShareRowExclusiveLock);
-	if (!bgw_job_stat_scan_job_id(bgw_job_id,
+	if (!bgw_job_stat_scan_job_id(job->fd.id,
 								  bgw_job_stat_tuple_mark_start,
 								  NULL,
 								  NULL,
 								  RowExclusiveLock))
-		bgw_job_stat_insert_relation(rel, bgw_job_id, true, DT_NOBEGIN);
+		bgw_job_stat_insert_relation(rel, job->fd.id, true, DT_NOBEGIN);
 	table_close(rel, NoLock);
+
+	/* We need to capture the execution start because failures are always logged */
+	job->job_history.execution_start = ts_timer_get_current_timestamp();
+	job->job_history.id = INVALID_BGW_JOB_STAT_HISTORY_ID;
+
+	if (ts_guc_enable_job_execution_logging)
```

Comment:
Nit: Since `ts_bgw_job_stat_history_mark_end` now performs the GUC check inside the function, I recommend moving the GUC check also inside `ts_bgw_job_stat_history_mark_start`. Otherwise, the functions behave differently; one has preconditions, and the other does not.

Reply:
I've reworked it a bit.